diff --git a/dist/mz_os.h b/dist/mz_os.h
index 4db0550..3b94cff 100644
--- a/dist/mz_os.h
+++ b/dist/mz_os.h
@@ -55,7 +55,20 @@ struct dirent {
 };
 typedef void *DIR;
 #else
+
+// patch: include dirent.h outside of the extern "C" block to fix Apple `modules-verifier` error:
+// > import of C++ module 'Darwin.POSIX.dirent' appears within extern "C" language linkage specification
+#ifdef __cplusplus
+}
+#endif
+
 #  include <dirent.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+// end patch
+
 #endif
 
 /***************************************************************************/
